src/include/optimizer/abstract_optimizer.h

```diff
@@ -43,6 +44,8 @@ class AbstractOptimizer {
       const std::unique_ptr<parser::SQLStatementList> &parse_tree) = 0;
 
   virtual void Reset(){};
+
+  concurrency::Transaction *txn = nullptr;
```

Comment:
I declare it in abstract optimizer because there was simple optimizer before. I have change it as you required but I don't know why you want me initialize in constructor. If someone add new constructor or if there are many constructor, it is kinda trouble than initialize here?